staging/src/k8s.io/component-base/logs/kube-log-runner/kube-log-runner.go

```diff
@@ -26,10 +26,14 @@ import (
 	"os/signal"
 	"strings"
 	"syscall"
+
+	"k8s.io/component-base/logs/logrotation"
 )
 
 var (
 	logFilePath    = flag.String("log-file", "", "If non-empty, save stdout to this file")
+	logFileSize    = flag.Uint("log-file-size", 0, "Useful with log-file, if non-zero, rotate log file when it reaches this size in MB")
```

Comment:
Is it possible for users to set the byte unit by themselves? For example, 10 MB or 10GB, instead of only having MB as a fixed unit? 😃

Reply:
for the file size to be zero, this is actually one of the issues this PR would like to tackle with, that the log file was not able to be refreshed/synced in time. 5s was added in this PR to refresh the logs, so there will be delay but it will be synced much quicker than the original one

Reply:
I'm trying to keep the parameters of kube-log-runner to be as simple as possible, but if we want more flexibility, yes, I can add that

Reply:
Ambiguous (size in B, KB, or MB?) command line flags are not simple because the user has to be very careful about using them correctly. If we do this, then we should do it properly, which means accepting a string for a resource.Quantity and parsing it accordingly.

The same argument applies to age: it should accept a `time.Duration`. There's even a `flag.Duration` for that.

Reply:
> This looks to me important for the case if the process loaded by "kube-log-runner" has a lot of parameters itself,

I don't understand this argument. Why should the number of arguments for the process influence what parameters get accepted by `kube-log-runner`?

Reply:
I guess what I mean is prefer the option of avoiding complicated parameters if we can achieve the similar functionality

Reply:
updated the log-file-size and log-file-age format, to make it more explicitly.